core/src/main/java/org/elasticsearch/action/bulk/BulkRequest.java

```diff
@@ -300,10 +300,16 @@ public BulkRequest add(BytesReference data, @Nullable String defaultIndex, @Null
                 if (token == null) {
                     continue;
                 }
-                assert token == XContentParser.Token.START_OBJECT;
+                if(token != XContentParser.Token.START_OBJECT) {
```

Comment:
Please use a space between the `if` and `(`. Also, I think it would be good to use `XContentParserUtils#ensureExpectedToken` or `XContentParserUtils#ensureFieldName`